### ISSUE TITLE
Dump Cluster - 8N2920980A M73 D06

### ISSUE THREAD

**Issue by cdgold3** [gmenounos/kw1281test#74]
Hi, I'm trying to Dump this cluster so I can retrieve the SKC but keep getting the following error:

`C:\kw1281test_0.97.0-beta_Win10>kw1281test.exe com4 9600 17 DumpMarelliMem 14336 2048
KW1281Test: Yesterday's diagnostics...Today.

Version 0.97.0-beta (https://github.com/gmenounos/kw1281test/releases)
Args: com4 9600 17 DumpMarelliMem 14336 2048
OSVersion: Microsoft Windows NT 10.0.22631.0
.NET Version: 8.0.0
Culture: en-GB
Opening serial port com4
Sending wakeup message
Reading sync byte
Keyword Lsb $01
Keyword Msb $8A
Protocol is KW 1281 (8N1)
ECU: 8N2920980A  KOMBI+WEGFAHRS. M73 D06
Software Coding 05444, Workshop Code: 00222
entryH: 0x00, regBlockH: 0x20, count: 0x0800
Sending block 0x6C
Writing data to cluster microcontroller
Receiving ACK
Caught: System.TimeoutException The operation has timed out.
Unhandled exception: System.TimeoutException: The operation has timed out.
   at System.IO.Ports.SerialPort.ReadByte()
   at BitFab.KW1281Test.Interface.GenericInterface.ReadByte() in /Users/gmenounos/src/kw1281test/Interface/GenericInterface.cs:line 34
   at BitFab.KW1281Test.KwpCommon.ReadByte() in /Users/gmenounos/src/kw1281test/KwpCommon.cs:line 130
   at BitFab.KW1281Test.Cluster.MarelliCluster.WriteMarelliBlockAndReadAck(Byte[]) in /Users/gmenounos/src/kw1281test/Cluster/MarelliCluster.cs:line 215
   at BitFab.KW1281Test.Cluster.MarelliCluster.DumpMem(String, UInt16, Nullable`1 ) in /Users/gmenounos/src/kw1281test/Cluster/MarelliCluster.cs:line 104
   at BitFab.KW1281Test.Cluster.MarelliCluster.DumpEeprom(Nullable`1, Nullable`1, String ) in /Users/gmenounos/src/kw1281test/Cluster/MarelliCluster.cs:line 21
   at BitFab.KW1281Test.Tester.DumpMarelliMem(UInt32, UInt32, ControllerInfo, String ) in /Users/gmenounos/src/kw1281test/Tester.cs:line 390
   at BitFab.KW1281Test.Program.Run(String[]) in /Users/gmenounos/src/kw1281test/Program.cs:line 341
   at BitFab.KW1281Test.Program.Main(String[]) in /Users/gmenounos/src/kw1281test/Program.cs:line 31`

For some reason the cluster restarts around the following command:

Writing data to cluster microcontroller
Receiving ACK

If i try ">kw1281test.exe com4 9600 17 DumpMarelliMem 3072 1024" i also get the same issue with the restart

**Comment by pcbbc** [comment]
Yep. The SKC code recovery doesn’t seem to work with the revised A version of the cluster.  You’ll notice that the cluster just reboots itself before sending the ACK.

I have both 8N2920980 and 8N2920980A on my bench, and the no-suffix version memory and SKC can be read just fine.  THE A suffix version on the other hand just bombs out.

My guess is they either fixed the exploit in the cluster firmware, or the exploit is relying on specific memory addresses in the firmware which are different in the A version.

**Comment by myphei** [comment]
Is there any way of getting the skc off this cluster or is it just not possible?

**Comment by gmenounos** [comment]
You might have luck with an R270+ device: https://github.com/gmenounos/kw1281test/wiki/Magneti-Marelli-Clusters
